Dockerfile

```diff
@@ -0,0 +1,33 @@
+# See https://github.com/rapidsai/ci-imgs for ARG options
+# NeMo Curator requires Python 3.10, Ubuntu 22.04/20.04, and CUDA 12 (or above)
+ARG CUDA_VER=12.5.1
+ARG LINUX_VER=ubuntu22.04
+ARG PYTHON_VER=3.10
+FROM rapidsai/ci-conda:cuda${CUDA_VER}-${LINUX_VER}-py${PYTHON_VER}
+
+WORKDIR /opt
+
+# Needed to navigate to and pull the forked repository's changes
+ARG FORKED_REPO_URL
+ARG CURATOR_COMMIT
```

Comment:
Please move these two lines below the expensive `RUN conda...` command.

Background:
Docker invalidates cache of `RUN` layers whenever a build-arg `ARG` was modified. This is, b/c behind the scenes docker mounts build-args as environment variables into each `RUN` process and so if the `ARG` changed the checksum of that layer differs to cache resulting in a cache break.

We want to define `ARG` as close as possible to the actual usage.

Better caching leads to not only quicker build-time but also to less storage pressure on the host

Reply:
Thank you! Updated.